File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/domain/ports/GiftCardRepository.java
package it.giannibombelli.wsc2026.giftcard.domain.ports;

import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCard;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

import java.util.Optional;

public interface GiftCardRepository {
    void save(GiftCard giftCard);

    Optional<GiftCard> findById(GiftCardId id);
}

File: typescript/src/giftcard/domain/ports/giftCardRepository.ts
import type { GiftCard } from '../giftcard/giftCard.js';
import type { GiftCardId } from '../giftcard/giftCardId.js';

export interface GiftCardRepository {
  save(giftCard: GiftCard): void;
  findById(id: GiftCardId): GiftCard | null;
}
